release-calendar/src/client/components/SearchBar.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import {ChangeEvent, FormEvent} from 'react';
+
+export interface SearchBarProps {
+  handleSearch: (searchValue: string) => void;
+}
+export class SearchBar extends React.Component<SearchBarProps, {}> {
+  constructor(props: Readonly<SearchBarProps>) {
+    super(props);
+    this.handleSearchValueChange = this.handleSearchValueChange.bind(this); //if I'm beinng honest, I'm not really
+    this.mySubmitHandler = this.mySubmitHandler.bind(this); //sure what the bindings actually accomplishes
+  }
+  mySubmitHandler = (event: FormEvent<HTMLFormElement>): void => {
```

Comment:
personal nit: you can just call these `submitHandler` and `searchValueChangeHandler`, `my` doesn't add anything to the name, and having them both use a consistent naming convention is nice

Reply:
haha, I didn't even realize I did that. Thanks for being so thorough. Will do!